dspace-api/src/main/java/org/dspace/ctask/general/BasicLinkChecker.java

```diff
@@ -132,10 +132,18 @@ protected int getResponseStatus(String url) {
         try {
             URL theURL = new URL(url);
             HttpURLConnection connection = (HttpURLConnection) theURL.openConnection();
-            int code = connection.getResponseCode();
-            connection.disconnect();
+            connection.setInstanceFollowRedirects(true);
+            int statusCode = connection.getResponseCode();
+            if ((statusCode == HttpURLConnection.HTTP_MOVED_TEMP || statusCode == HttpURLConnection.HTTP_MOVED_PERM ||
+                    statusCode == HttpURLConnection.HTTP_SEE_OTHER)) {
+                connection.disconnect();
+                String newUrl = connection.getHeaderField("Location");
+                if (newUrl != null) {
+                    return getResponseStatus(newUrl);
```

Comment:
Because this is recursive, we need to set a limit on the number of times it might redirect...otherwise, this link checker will hang indefinitely if it encounters an invalid/circular redirect.

Reply:
Changed it so this branch now points to main and implemented the fix you mentioned with a new configuration in curate.cfg. I have it set to 0 by default currently and -1 allows for unlimited redirects and explained that in the file.